### PR TITLE
ci: circle migration, build test, prepare and prepublishOnly

### PR DESCRIPTION
**Pull request by atlawrie**
Migrates our CI pipeline from Travis to CircleCI. Adds an additional test step to our CI process to test successful builds. Adds additional scripts to ensure properly updated `dist` prior to release.

### CODE REVIEW

File: .travis.yml
Comment (sherwinski) [inline .travis.yml:11]: Same point as the other PR about adding this line in

**Review by sherwinski — APPROVED** [review]
LGTM but I was wondering if it'd be worth squashing some of these commits down before merging? Seems like there are a handful of commits that overwrite each other

File: .travis.yml
Comment (atlawrie) [reply .travis.yml:11]: Added the lint step, and will squash the commits in the merge.

**Pull request merged**
